src/operator/aggregate/mod.rs

```diff
@@ -39,13 +39,42 @@ pub use min::Min;
 /// A trait for aggregator objects.  An aggregator summarizes the contents
 /// of a Z-set into a single value.
 ///
+/// This trait supports two aggregation methods that can be combined in
+/// various ways to compute the aggregate efficiently.  First, the
+/// [`aggregate`](`Self::aggregate`) method takes a cursor pointing to
+/// the first key of a Z-set and scans the cursor to compute the aggregate
+/// over all values in the cursor.  Second, the
+/// [`Semigroup`](`Self::Semigroup`) associated type allows aggregating
+/// partitioned Z-sets by combining aggregates computed over individual
+/// partitions (e.g., computed by different worker threads).
+///
+/// This design requires aggregate values to form a semigroup with a `+`
+/// operation such that `Agg(x + y) = Agg(x) + Agg(y)`, i.e., aggregating
+/// a union of two Z-sets must produce the same result as the sum of
+/// individual aggregates.  Not all aggregates have this property.  E.g.,
+/// the average value of a Z-set cannot be computed by combining the
+/// averages of its subsets.  We can get around this problem by representing
+/// average as `(sum, count)` tuple with point-wise `+` operator
+/// `(sum1, count1) + (sum2, count2) = (sum1+sum2, count1+count2)`.
+/// The final result is converted to an actual average value by dividing
+/// the first element of the tuple by the second.  This is a general technique
```

Comment:
for some aggregators, such as exact median, it may need to simply keep the full set around.